lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -106,6 +108,22 @@ def write_playbook
         File.write(File.join(@root, 'project', 'playbook.yml'), @playbook)
       end
 
+      def write_ssh_key
+        key_path = File.join(@root, 'env', 'ssh_key')
+        File.symlink(ForemanRemoteExecutionCore.settings[:ssh_identity_key_file], key_path)
+
+        if !@passphrase.nil? && !@passphrase.empty?
+          passwords_path = File.join(@root, 'env', 'passwords')
+          secrets = <<~SECRETS
+            ---
+            "for.*/artifacts/.*/ssh_key_data:": "#{@passphrase}"
+          SECRETS
+          File.write(passwords_path, '')
+          File.chmod(0o0600, passwords_path)
+          File.write(passwords_path, secrets)
```

Comment:
You need to *create* the file with the right permissions, as otherwise an attacker just needs to wait for your create to happen, acquire a file handle (at this moment the permissions might be OK), and wait for you to write data to it.
The chmod won't have any influence on the already open handle.